pulp_rpm/app/tasks/publishing.py

```diff
@@ -516,10 +521,17 @@ def generate_repo_metadata(
     pkg_pks_to_ignore = set()
     latest_build_time_by_nevra = defaultdict(list)
     packages = Package.objects.filter(pk__in=content)
-    for pkg in packages.only(
-        "pk", "name", "epoch", "version", "release", "arch", "time_build"
+    for pkg in packages.values(
+        "pk",
+        "name",
+        "epoch",
+        "version",
+        "release",
+        "arch",
+        "time_build",
     ).iterator():
-        latest_build_time_by_nevra[pkg.nevra].append((pkg.time_build, pkg.pk))
+        nevra = format_nevra(pkg["name"], pkg["epoch"], pkg["version"], pkg["release"], pkg["arch"])
+        latest_build_time_by_nevra[nevra].append((pkg["time_build"], pkg["pk"]))
```

Comment:
I could fix this by just including the `pulp_type` in the list of values, but I  decided to just ditch using the `Package` class here in the first place